contracts/KeysManager.sol

```diff
@@ -402,6 +412,7 @@ contract KeysManager is EternalStorage, IKeysManager {
         _setMiningKeyByPayout(payoutKey, address(0));
         _clearMiningKey(_key);
         _getValidatorMetadata().clearMetadata(_key);
+        _setHasMiningKeyBeenRemoved(_key);
```

Comment:
Let's also call `_setHasMiningKeyBeenRemoved` inside the `swapMiningKey` function because the swapped key should also be considered as removed.